src/harness/util.py

```diff
@@ -279,3 +279,129 @@ def convertRequestToRequestWithCpiSignature(private_key, cpi_id,
+    from reference_models.antenna import antenna
+    from reference_models.geo import vincenty
+    from reference_models.geo import nlcd
+    TERRAIN_TEST_DIR = os.path.join(os.path.dirname(__file__), 'reference_models', 'geo', 'testdata', 'ned')
+    NLCD_TEST_DIR = os.path.join(os.path.dirname(__file__), 'reference_models', 'geo', 'testdata', 'nlcd')
+    ITU_TEST_DIR = os.path.join(os.path.dirname(__file__), 'reference_models', 'geo', 'testdata', 'itu')
+    wf_itm.ConfigureTerrainDriver(terrain_dir=TERRAIN_TEST_DIR)
+    wf_itm.ConfigureItuDrivers(itu_dir=ITU_TEST_DIR)
+    wf_hybrid.ConfigureTerrainDriver(terrain_dir=TERRAIN_TEST_DIR)
+    wf_hybrid.wf_itm.ConfigureItuDrivers(itu_dir=ITU_TEST_DIR)
+    nlcd_driver = nlcd.NlcdDriver(NLCD_TEST_DIR)
+
+    request = json.loads(requestJson)
+    reliabilityLevel = request['reliabilityLevel']
+    if reliabilityLevel not in [-1, 0.05, 0.95]:
+        response = 404
+        return response
+    Tx = request['cbsd']
+    if ('fss' in request) and ('ppa' in request):
+        response = 404
+        return response
+    elif 'ppa' in request:
+        Rx ={}
+        Rx['height'] = 1.5
+        isfss = False
+        coordinates = [];
+        ppa = request['ppa']
+        boundarysPoints = ppa['geometry']['coordinates']
+        boundarypoints = []
+        for BP in boundarysPoints:
+            boundarypoints.append([BP[1], BP[0]])
+        
+        regionVal = nlcd_driver.RegionNlcdVote(boundarypoints)
+        ARCSEC = 2
+        res, lat, lon = getPPinPPA(boundarysPoints, ARCSEC)
+        if res== 0:
+            Rx['latitude'] = lat
+            Rx['longitude']= lon
+        else:
+            response = 404
+            return response
+
+    elif 'fss' in request:
+        isfss = True
+        Rx = request['fss']
+    else:
+        response = 404
+        return response
+    
+# ITM pathloss (if receiver type is FSS) or the hybrid model pathloss (if receiver type is PPA) and corresponding antenna gains.
+# The test specification notes that the SAS UUT shall use default values for w1 and w2 in the ITM model.
+    Result = {}
+    
+    d, az, rev_az = vincenty.GeodesicDistanceBearing(Tx['latitude'], Tx['longitude'], Rx['latitude'], Rx['longitude'])
+
+    gainTxRx = antenna.GetStandardAntennaGains(az, ant_azimuth=Tx['antennaAzimuth'], ant_beamwidth=Tx['antennaBeamwidth'], ant_gain=Tx['antennaGain'])
+    Result['txAntennaGainDbi'] = gainTxRx   
+
+    if isfss:
+        PathLoss = wf_itm.CalcItmPropagationLoss(Tx['latitude'], Tx['longitude'], Tx['height'], Rx['latitude'], Rx['longitude'], Rx['height'], reliability=reliabilityLevel, freq_mhz=3625.)
+        Result['pathlossDb'] = PathLoss.db_loss
+      
+    else:
+        
+        PathLoss = wf_hybrid.CalcHybridPropagationLoss(Tx['latitude'], Tx['longitude'], Tx['height'], Rx['latitude'], Rx['longitude'], Rx['height'], reliability=-1, freq_mhz=3625., region=regionVal)
+        Result['pathlossDb'] = PathLoss.db_loss
+ 
```

Comment:
Here: if Rx is a dict, no need to do `in Rx.keys()`.
You can simply do:
`if 'rxAntennaGainRequired' in Rx:`